integration_tests/backend/test_backend.py

```diff
@@ -0,0 +1,73 @@
+import pytest
+import requests
+import subprocess
+from pathlib import Path
+import os
+import sys
+
+import aqueduct
+
+
+def get_response(endpoint, additional_headers={}):
+    headers = {"api-key": pytest.api_key}
+    headers.update(additional_headers)
+    url = f"{pytest.server_address}{endpoint}"
+    if url[:4] != "http":
+        try:
+           r = requests.get("https://"+url, headers=headers) 
+        except:
+            try:
+                r = requests.get("http://"+url, headers=headers) 
+            except:
+                raise Exception(f"Cannot connect to {url}")
```

Comment:
Would it be possible to use the same logical structure that we use in the api_client.py on the SDK? That way its 1) more consistent across our tests and 2) avoiding having nested try-except clauses which is a bit confusing.

Reply:
Actually, should I use the `use_http` variable directly?
<img width="567" alt="image" src="https://user-images.githubusercontent.com/30596854/179851167-638596f5-a701-4a58-9841-fa4c54143ca3.png">

Reply:
hmm that might be better - how about we add a method on api_client called `ui_and_server_address()`, which will return the {prefix}{server_address} as a string that you can join with the routes? This'll be a useful internal method anyways  - see `generate_ui_url()` in the SDK's utils.py

Reply:
I like your suggestion of keeping the logic in one place, this just wraps it a little more.

Reply:
- using existing `api_client._construct_full_url` in tests
- created `api_client._construct_base_url`
- `api_client._construct_full_url` uses `api_client._construct_base_url`
- made the arg `use_http` optional for`api_client._construct_full_url` and `api_client._construct_base_url` (if `use_http=None`, use `self.use_http`)
- refactor `generate_ui_url` to use `api_client._construct_full_url`
- clean url in `api_client` so `SERVER_ADDRESS=localhost:8080/` works (would error out before)